### ISSUE TITLE
TooltipTrigger: Using Esc key to dismiss a Tooltip within a Dialog closes the Dialog.

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#4120]
# 🐛 Bug Report 

## 🤔 Expected Behavior

Using the Esc key to dismiss a Tooltip within a Dialog should not also close the Dialog.

## 😯 Current Behavior

Using Esc key to dismiss a Tooltip within a Dialog closes the Dialog. Here is a [Code Sandbox example](https://codesandbox.io/s/tooltip-closes-parent-dialog-on-esc-m6j08h?file=/src/App.js) that illustrates the problem. 

## 🔦 Context

Issue was raised in context of a Dialog containing elements containing truncated text that display the full text using a Tooltip. A keyboard user cannot dismiss the Tooltip without dismissing the Dialog.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | [@adobe/react-spectrum@3.25.0](https://github.com/adobe/react-spectrum/releases/tag/%40adobe%2Freact-spectrum%403.25.0)
| Browser          | Chrome Version 110.0.5481.177 (Official Build) (arm64)
| Operating System | macOS Ventura 13.2.1 (22D68)

**Comment by LFDanLu** [comment]
Related: https://github.com/adobe/react-spectrum/issues/3987. I'll keep this one open as well for now so we can track what components will need this fix.
